modules/core/src/main/java/org/apache/ignite/internal/cluster/IgniteClusterImpl.java

```diff
@@ -315,15 +317,27 @@ public IgniteClusterImpl(GridKernalContext ctx) {
 
         try {
             ctx.state().changeGlobalState(active, baselineNodes(), false).get();
+
+            recordEvent(active ? EventType.EVT_CLUSTER_ACTIVATED : EventType.EVT_CLUSTER_DEACTIVATED);
         }
         catch (IgniteCheckedException e) {
+            recordEvent(EventType.EVT_CLUSTER_ACTIVATION_FAILED);
+
             throw U.convertException(e);
         }
         finally {
             unguard();
         }
     }
 
+    /** */
+    private void recordEvent(int evtType) {
+        if (ctx.event().isRecordable(evtType)) {
+            ClusterActivationEvent evt = new ClusterActivationEvent(null, null, evtType);
```

Comment:
Please provide local node (ctx.discovery().localNode()) and human readable message describing event. We do that for all the other events so it makes sense here. Also please add empty line after this one.